### ISSUE TITLE
Unity iOS AOT error serializing a dictionary

### ISSUE THREAD

**Issue by scatterbrain** [msgpack/msgpack-cli#148]
I'm trying to serialize a Dictionary<string, object> which works well in Unity editor but fails with AOT error when running on iOS.

```c#
	public byte[] Serialize()
	{
		using (var byteStream = new MemoryStream()) {
			var packer = Packer.Create (byteStream);
			packer.PackDictionary<string, object> (data);
			return byteStream.ToArray();
		}
	}
```

The error I get: 

```
ExecutionEngineException: Attempting to call method 'MsgPack.Serialization.SerializationContext::GetSerializer' for which no ahead of time (AOT) code was generated.
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at MsgPack.Serialization.ReflectionExtensions.InvokePreservingExceptionType (System.Reflection.MethodInfo source, System.Object instance, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at MsgPack.Serialization.SerializationContext+SerializerGetter.Get (MsgPack.Serialization.SerializationContext context, System.Type targetType, System.Object providerParameter) [0x00000] in <filename unknown>:0 
  at MsgPack.PackerUnpackerExtensions.PackObjectCore (MsgPack.Packer source, System.Object value, MsgPack.Serialization.SerializationContext context) [0x00000] in <filename unknown>:0 
  at MsgPack.PackerUnpackerExtensions.PackObject (MsgPack.Packer source, System.Object value, MsgPack.Serialization.SerializationContext context) [0x00000] in <filename unknown>:0 
  at MsgPack.PackerUnpackerExtensions.PackDictionaryCore[TKey,TValue] (MsgPack.Packer source, IDictionary`2 dictionary, MsgPack.Serialization.MessagePackSerializer`1 keySerializer, MsgPack.Serialization.MessagePackSerializer`1 valueSerializer) [0x00000] in <filename unknown>:0 
  at MsgPack.PackerUnpackerExtensions.PackDictionary[TKey,TValue] (MsgPack.Packer source, IDictionary`2 dictionary) [0x00000] in <filename unknown>:0 
  at NetworkEvent.Serialize () [0x00000] in <filename unknown>:0 
```

Unity 5.3.1f1
msgpack-cli 0.6.8 unity-full/MsgPack.dll

Any help is greatly appreciated!

**Comment by yfakariya** [comment]
Thank you for reporting.
It should be caused by `object` type, so you can avoid this exception with another type instead of `object`.
That aside, I will fix this AOT error.

**Comment by yfakariya** [comment]
@scatterbrain This kind of AOT error should be fixed in 0.7.0-alpha3. Would you confirm it?

**Comment by yfakariya** [comment]
This issue should be solved in 0.7.
Feel free to reopen if you find any issues.
